doc/source/data/performance-tips.rst

```diff
@@ -189,3 +134,79 @@ setting the ``DataContext.use_push_based_shuffle`` flag:
         ray.data.range(1000)
         .random_shuffle()
     )
+
+Configuring execution
+---------------------
+
+Configuring resources and locality
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the CPU and GPU limits are set to the cluster size, and the object store memory limit conservatively to 1/4 of the total object store size to avoid the possibility of disk spilling.
+
+You may want to customize these limits in the following scenarios:
+- If running multiple concurrent jobs on the cluster, setting lower limits can avoid resource contention between the jobs.
+- If you want to fine-tune the memory limit to maximize performance.
+- For data loading into training jobs, you may want to set the object store memory to a low value (e.g., 2GB) to limit resource usage.
+
+Execution options can be configured via the global DataContext. The options will be applied for future jobs launched in the process:
+
+.. code-block::
+
+   ctx = ray.data.DataContext.get_current()
+   ctx.execution_options.resource_limits.cpu = 10
+   ctx.execution_options.resource_limits.gpu = 5
+   ctx.execution_options.resource_limits.object_store_memory = 10e9
+
+
+Locality with output (ML ingest use case)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   ctx.execution_options.locality_with_output = True
+
+Setting this to True tells Ray Data to prefer placing operator tasks onto the consumer node in the cluster, rather than spreading them evenly across the cluster. This can be useful if you know you'll be consuming the output data directly on the consumer node (i.e., for ML training ingest). However, this may incur a performance penalty for other use cases.
+
+Reproducibility
+---------------
+
+Deterministic execution
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   # By default, this is set to False.
+   ctx.execution_options.preserve_order = True
+
+To enable deterministic execution, set the above to True. This may decrease performance, but will ensure block ordering is preserved through execution. This flag defaults to False.
+
+.. _datasets_pg:
+
+Ray Data and placement groups
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Does this belong under reproducibility? It feels more like scheduling details and less of a performance tip